src/core/qgsvectorlayereditutils.cpp

```diff
@@ -475,35 +475,27 @@ QgsGeometry::OperationResult QgsVectorLayerEditUtils::splitParts( const QgsPoint
     fit = mLayer->getFeatures( QgsFeatureRequest().setFilterRect( bBox ).setFlags( QgsFeatureRequest::ExactIntersect ) );
   }
 
-  QgsGeometry::OperationResult addPartRet = QgsGeometry::OperationResult::Success;
-
   QgsFeature feat;
   while ( fit.nextFeature( feat ) )
   {
     QVector<QgsGeometry> newGeometries;
     QgsPointSequence topologyTestPoints;
     QgsGeometry featureGeom = feat.geometry();
     splitFunctionReturn = featureGeom.splitGeometry( splitLine, newGeometries, topologicalEditing, topologyTestPoints );
+
     if ( splitFunctionReturn == 0 )
     {
-      //add new parts
-      if ( !newGeometries.isEmpty() )
-        featureGeom.convertToMultiType();
+      QgsGeometry newGeom( newGeometries.at( 0 ) );
```

Comment:
not sure if it's checked before, but this might crash if newGeometries is empty.

Reply:
Normally with splitFunctionReturn, but I added a test if newGeometries is empty. Thanks.